logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/call/usecase/EndCallOnConversationChangeUseCaseTest.kt

```diff
@@ -27,97 +29,133 @@ import io.mockative.mock
+        }
 
-    @Mock
-    private val endCall = mock(classOf<EndCallUseCase>())
+        endCallOnConversationChange()
 
-    private lateinit var endCallOnConversationChange: EndCallOnConversationChangeUseCase
+        verify(arrangement.endCall)
+            .suspendFunction(arrangement.endCall::invoke)
+            .with(eq(conversationId))
+            .wasInvoked(once)
+    }
 
-    @BeforeTest
-    fun setup() {
-        endCallOnConversationChange = EndCallOnConversationChangeUseCaseImpl(
-            callRepository = callRepository,
-            conversationRepository = conversationRepository,
-            endCallUseCase = endCall
-        )
+    @Test
+    fun givenAnEstablishedCall_whenUserIsRemovedFromConversation_thenEndTheCurrentCall() = runTest {
+        val (arrangement, endCallOnConversationChange) = arrange {
+            withObserveConversationDetailsByIdReturning(Either.Right(groupConversationDetail))
+        }
 
-        given(callRepository)
-            .suspendFunction(callRepository::establishedCallsFlow)
-            .whenInvoked()
-            .thenReturn(flowOf(listOf(call)))
+        endCallOnConversationChange()
 
-        given(endCall)
-            .suspendFunction(endCall::invoke)
-            .whenInvokedWith(eq(conversationId))
-            .thenDoNothing()
+        verify(arrangement.endCall)
+            .suspendFunction(arrangement.endCall::invoke)
+            .with(eq(conversationId))
+            .wasInvoked(once)
     }
 
     @Test
-    fun givenAnEstablishedCall_whenConversationIsDeleted_thenEndTheCurrentCall() = runTest {
-
-        given(conversationRepository)
-            .suspendFunction(conversationRepository::observeConversationDetailsById)
-            .whenInvokedWith(eq(conversationId))
-            .then {
-                flowOf(Either.Left(StorageFailure.DataNotFound))
-            }
+    fun givenAnEstablishedCall_whenUserDeletesHisAccount_thenEndTheCurrentCall() = runTest {
+        val (arrangement, endCallOnConversationChange) = arrange {
+            withObserveConversationDetailsByIdReturning(Either.Right(oneOnOneConversationDetail))
+        }
 
         endCallOnConversationChange()
 
-        verify(endCall)
-            .suspendFunction(endCall::invoke)
+        verify(arrangement.endCall)
+            .suspendFunction(arrangement.endCall::invoke)
             .with(eq(conversationId))
             .wasInvoked(once)
     }
 
     @Test
-    fun givenAnEstablishedCall_whenUserIsRemovedFromConversation_thenEndTheCurrentCall() = runTest {
-
-        given(conversationRepository)
-            .suspendFunction(conversationRepository::observeConversationDetailsById)
-            .whenInvokedWith(eq(conversationId))
-            .then {
-                flowOf(Either.Right(groupConversationDetail))
-            }
+    fun givenAnEstablishedCall_whenConversationProteusDegraded_thenEndTheCurrentCall() = runTest {
```

Comment:
suggestion: I know this pr mostly about Proteus, but since the code covers MLS verification status, I suggest to have a test for MLS degraded end call as well.